src/image.cpp

```diff
@@ -271,6 +271,32 @@ void ARGBImage::writeBMP( std::ostream& streamout ) {
     delete[] imageData;
 }
 
+void ARGBImage::writeWEBP( std::ostream& streamout, int quality ) {
+    // assume 4 channels
+    if (channels != 4) {
+        throw new std::runtime_error("WebP tried to save image with more than 4 channels");
+    }
+
+    size_t size;
+    uint8_t* out;
+    if (quality == 10) {
+        // encode lossless at highest quality
+        size = WebPEncodeLosslessRGBA(data, width, height, width * 4, &out);
```

Comment:
Actually, there is more advanced interface for lossless encoding:

```c
struct WebPConfig {
  int lossless;           // Lossless encoding (0=lossy(default), 1=lossless).
  float quality;          // between 0 and 100. For lossy, 0 gives the smallest
                          // size and 100 the largest. For lossless, this
                          // parameter is the amount of effort put into the
                          // compression: 0 is the fastest but gives larger
                          // files compared to the slowest, but best, 100.
  int method;             // quality/speed trade-off (0=fast, 6=slower-better)
```

It allows us to specify encoding speed for webp in lossless mode. I am not sure if we should use this or just stick to default prefubs.

Reply:
I saw this as well, but I think we have no good reason to use other settings, unless the user can change them; but that would require more switches, which I think would introduce unnecessary complexity, which is also why I think re-using `--quality` for deciding between lossy/lossless encoding is good, compared to adding something like `--lossless-webp`.